src/Microsoft.AspNet.OData/Formatter/Deserialization/ODataDeserializerContext.cs

```diff
@@ -27,6 +28,11 @@ public HttpRequestMessage Request
                 _request = value;
                 InternalRequest = _request != null ? new WebApiRequestMessage(_request) : null;
                 InternalUrlHelper = _request != null ? new WebApiUrlHelper(_request.GetUrlHelper()) : null;
+
+                var request = (WebApiRequestMessage)InternalRequest;
```

Comment:
>var request = (WebApiRequestMessage)InternalRequest; [](start = 16, length = 52)

why don't use the "value" here?

Reply:
besides, "InternalRequest" could be null at line29, right?

---
In reply to: [501472502](https://github.com/OData/WebApi/pull/2311#discussion_r501472502) [](ancestors = 501472502)

Reply:
If `InternalRequest` is `null`, `request` will be `null` and `EnableCaseInsensitiveModelBinding` will be `false`

Reply:
Let me rename `request` to `webApiRequestMessage`.

Reply:
#Resolved

Reply:
This still seems suspect to me.  

This is a public API, so the user could presumably call set with _request = null. As Sam says, we handle that case in line 29, but line 35 would throw a null ref exception when we tried to deref the Configuration property of the (null) webApiRequestMessage.

---
In reply to: [501514115](https://github.com/OData/WebApi/pull/2311#discussion_r501514115) [](ancestors = 501514115)